### DISCUSSION TITLE
Send e-mail without implementing a Mailable class

### DISCUSSION THREAD

**Discussion by aimeos** [laravel/framework#41632] [General]
We are currently updating the Aimeos package for Laravel 9. In Laravel 8 and before, it was possible to send an e-mail without implementing a Mailable class by accessing the SwiftMailer object directly. In Laravel 9, only the Symfony Mail Transport class is returned, which doesn't have a `send()` method. Is there still a way to send e-mail without implementing a Mailable like it's still possible according to the Symfony 6 mail example:

```
use Symfony\Component\Mailer\Transport;
use Symfony\Component\Mailer\Mailer;

$transport = Transport::fromDsn('smtp://localhost');
$mailer = new Mailer($transport);

$email = (new Email())
    ->from('hello@example.com')
    ->to('you@example.com')
    //->cc('cc@example.com')
    //->bcc('bcc@example.com')
    //->replyTo('fabien@example.com')
    //->priority(Email::PRIORITY_HIGH)
    ->subject('Time for Symfony Mailer!')
    ->text('Sending emails is fun again!')
    ->html('<p>See Twig integration for better HTML integration!</p>');

$mailer->send($email);
```

What we would need is access to the `Symfony\Component\Mailer\Mailer` class or the corresponding Laravel class wrapping this object and which offers a `send()`-like method.

**Comment by chuckrincon** [comment]
Hi there ✋🏻 

You could try this:

```php
    Mail::send([], [], function (Illuminate\Mail\Message $message) {
        $message
            ->from('hello@example.com')
            ->to('you@example.com')
            ->cc('cc@example.com')
            ->bcc('bcc@example.com')
            ->replyTo('fabien@example.com')
            ->subject('Time for Symfony Mailer!')
            ->text('Sending emails is fun again!')
            ->html('<p>See Twig integration for better HTML integration!</p>');
    });
```
where `Mail::send` will be your `Symfony\Component\Mailer\Mailer` class wrapping  and the `$message` argument will be your `Symfony\Component\Mime\Email` class instance

**Reply by aimeos** [reply]
Thanks for your suggestion but it's not really better than implementing a Mailable class because of the closure you need to pass. Instead of passing the message to the `send()` method, you get a message object which you must populate within the closure. This is problematic if you already have a Message object which got populated before.

Our current workaround is:
```
$message = new \Symfony\Component\Mime\Email();
// populate message object
app( 'mailer' )->getSymfonyTransport()
    ->send( $message, \Symfony\Component\Mailer\Envelope::create( $message ) );
```

We are now using the Symfony objects only which is not perfect because we would like to use the Laravel objects instead. The best solution would be if the `createMessage()` method of the Larvel Mailer object would be public and a `sendMessage()` method is added which accepts a Laravel message object.

When it comes to such customizations, Laravel 9 mail implementation is worse than Laravel 8 and before. The Laravel 9 mail implementation is highly opinionated and only works the way it is meant to. Also, the queue and view object dependencies doesn't feel right regarding the "separation of concerns" principle.
